File: 2460_apply_operation_to_an_array/main.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function applyOperations($nums) {
        $zero_indexes = [];

        for ($i=0; $i<count($nums)-1; $i++) {

            if ($nums[$i] == 0) {
                $zero_indexes[] = $i;
                continue;
            }

            if ($nums[$i] == $nums[$i+1]) {

                if (count($zero_indexes)) {
                    $nums[array_shift($zero_indexes)] = 2*$nums[$i];
                    $nums[$i+1] = 0;
                    $nums[$i] = 0;
                    $zero_indexes[] = $i;
                }
                else {
                    $nums[$i] = 2*$nums[$i];
                    $nums[$i+1] = 0;
                }
            }
            else {
                if (count($zero_indexes)) {
                    $nums[array_shift($zero_indexes)] = $nums[$i];
                    $nums[$i] = 0;
                    $zero_indexes[] = $i;
                }
            }
        }

        if (count($zero_indexes)) {
            $nums[array_shift($zero_indexes)] = $nums[count($nums) - 1];
            $nums[count($nums) - 1] = 0;
        }

        return $nums;
    }
}

$sol = new Solution();
$nums = [1, 2, 5, 6, 6, 9, 0, 0, 1];
print_r($sol->applyOperations($nums));

File: 2460_apply_operation_to_an_array/main.py
from typing import List


class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:

        zero_indexes = []

        for i in range(len(nums) - 1):

            if nums[i] == 0:
                zero_indexes.append(i)
                continue

            if nums[i] == nums[i+1]:
                if len(zero_indexes) > 0:
                    nums[zero_indexes.pop(0)] = nums[i] * 2
                    nums[i+1] = 0
                    nums[i] = 0
                    zero_indexes.append(i)
                else:
                    nums[i] = nums[i] * 2
                    nums[i+1] = 0
            else:
                if len(zero_indexes) > 0:
                    nums[zero_indexes.pop(0)] = nums[i]
                    nums[i] = 0
                    zero_indexes.append(i)

        if len(zero_indexes) > 0:
            nums[zero_indexes.pop(0)] = nums[-1]
            nums[-1] = 0

        return nums
    

sol = Solution()
nums = [847,847,0,0,0,399,416,416,879,879,206,206,206,272]

print(sol.applyOperations(nums))